pkg/services/ngalert/api/api_ruler_export_test.go

```diff
@@ -253,12 +252,16 @@ func TestExportRules(t *testing.T) {
 		))
 	ruleStore.PutRule(context.Background(), hasAccess2...)
 
-	_, noAccess2 := ngmodels.GenerateUniqueAlertRules(10,
+	_, noAccessByFolder := ngmodels.GenerateUniqueAlertRules(10,
 		ngmodels.AlertRuleGen(
 			ngmodels.WithUniqueUID(&uids),
 			ngmodels.WithQuery(accessQuery), // no access because of folder
+			ngmodels.WithNamespaceUIDNotIn(f1.UID, f2.UID),
 		))
-	ruleStore.PutRule(context.Background(), noAccess2...)
+
+	ruleStore.PutRule(context.Background(), noAccessByFolder...)
+	// overwrite the folders visible to user because PutRule automatically creates folders.
```

Comment:
```suggestion
	// overwrite the folders visible to user because PutRule automatically creates folders in production.
```

Reply:
I think this is only true for the test rule store, not the production store.

Reply:
Yeah you are right, I looked at the wrong implementation.